mutacc/utils/bam_parse.py

```diff
@@ -3,23 +3,33 @@
 
 import pysam
 
-def getOverlapingReads(fileName, chrom, start, n=0):
+def get_overlaping_reads(fileName, chrom, start, end, padding = 300):
+"""
+
+Extracts all read names from a bam file, overlapping the specified region.
+
+Args:
+
+    fileName (string): name of bam file
+    chrom (string): name of chromosome
+    start (int): start of region
+    end (int): end of region
+    padding (int): optional padding of region. i.e. Reads in the vicinity vill also be detected
+    depending on the padding parameter. 
+
+Returns:
+
+    reads (list): List of read identifiers. 
     
-    with pysam.AlignmentFile(fileName, 'rb') as file:
+"""
+    sam_file = pysam.AlignmentFile(fileName, 'rb')
         
-        reads = file.fetch(chrom, start - n, start + n)
+    reads = sam_file.fetch(reference = chrom, 
+            start = start - padding, 
+            end = end + padding)
     
     return [read.query_name for read in reads]
```

Comment:
To make this function more light weigth you could make this as a generator with the yield statement like

```python
for read in reads:
    yield read.query_name
```

Than any function that calls this can iterate the result just like a list. If there are many reads expected this is a good solution